emissionsapi/web.py

```diff
@@ -70,6 +73,15 @@ def get_data(session, country=None, geoframe=None, begin=None, end=None):
         if country not in country_bounding_boxes:
             return 'Unknown country code.', 400
         rectangle = bounding_box_to_wkt(*country_bounding_boxes[country][1])
+    # parse parameter wkt
+    elif polygon is not None:
+        if len(polygon) % 2 != 0:
+            return ('The number of parameter is not even! Check if you have'
+                    ' matching lon-lat coordinate pairs', 400)
+        try:
+            rectangle = polygon_to_wkt(polygon)
+        except ValueError:
```

Comment:
Maybe
```
        except ValueError as err:
            return str(e), 400
```

Reply:
Thats nice! And it renders the dedicated response message obsolete!

Reply:
This is a nice solution, but I would be careful with returning the raised Exception directly to the API. This could lead to information leakage. There could be a `ValueError` we do not want to forward to the user. See for example https://www.owasp.org/index.php/Error_Handling about information leakage during error handling.

You could circumvent that by raising a self-defined exception and only catch that.

Reply:
I thought about that when suggesting this and I gambled on a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) defined as *„Raised when an operation or function receives an argument that has the right type but an inappropriate value“* shouldn't actually reveal any critical internal secrets here.

That said I fine with both, working with a custom error or just using a generic error message.

Reply:
Do I create an own exception class in a new file or add it to the utils.py?

Reply:
Since it will likely be used just in here, I would define this at the top of this file. Especially since the definition should be rather short:

```python
class RESTParamError(ValueError):
     pass
```

Reply:
@lkiesow like [that](https://github.com/emissions-api/emissions-api/pull/108/files#diff-22cbaadb49831511ce93e6c62e59381c)?